agent/agent.go

```diff
@@ -517,6 +517,15 @@ func (a *Agent) runOutputs(
 		go func(output *models.RunningOutput) {
 			defer wg.Done()
 
+			if flushImmediately {
+				// wait long enough for one input collection, then flush.
+				if err := internal.SleepContext(ctx, firstIntervalCollectionTime+1*time.Second); err != nil {
```

Comment:
Sleeping for interval+jitter+1 feels too racy.  If we're going to add this, would it make sense to gather from inputs without timeout?

Reply:
It's absolutely race-y. Especially if you want to take interval out of the calculation and replace it with the_longest_avg_runtime_i_really_want_to_reasonably_wait_for_all_inputs_to_complete (worst or best variable name ever?).
The outputs have no context for how long to wait for a single round of all inputs to complete. As far as i know there's no good answer to that, and if you're going to use this feature you're going to have to either: 
1. accept a disclaimer that you might miss metrics
2. set your own time you feel is reasonable to wait until the first metric collection (but #1 is also true)